analyzers/tests/SonarAnalyzer.UnitTest/TestFramework/ParseOptionsHelper.cs

```diff
@@ -94,9 +94,9 @@ language switch
             };
 
         private static ImmutableArray<ParseOptions> FilterByEnvironment(this IEnumerable<ParseOptions> options) =>
-            TestContextHelper.IsAzureDevOpsContext
-                ? options.ToImmutableArray()
-                : ImmutableArray.Create(options.First()); // Use only the oldest version for local test run and debug
+            TestContextHelper.IsLocalOrPullRequestBuild
```

Comment:
_Open question:_ This works, but wouldn't this be better? To keep the helper functions atomic.
```suggestion
            TestContextHelper.IsAzureDevOpsContext && !TestContextHelper.IsPullRequestBuild
```